packages/react-sdk/src/core/hooks/useDevices.ts

```diff
@@ -1,11 +1,50 @@
-import { useEffect, useState } from 'react';
+import { ChangeEvent, useEffect, useState } from 'react';
 import { Observable, pairwise } from 'rxjs';
 import {
   getAudioDevices,
   getAudioOutputDevices,
   getVideoDevices,
 } from '@stream-io/video-client';
 
+export const useCheckBrowserPermissions = (permissionName: PermissionName) => {
+  const [canSubscribe, enableSubscription] = useState(false);
+
+  useEffect(() => {
+    let permissionState: PermissionStatus;
+    const handlePermissionChange = (e: Event) => {
+      const { state } = (e as unknown as ChangeEvent<PermissionStatus>).target;
+      if (state === 'granted') {
+        enableSubscription(true);
+      } else {
+        enableSubscription(false);
+      }
+    };
+    const checkPermissions = async () => {
+      try {
+        permissionState = await navigator.permissions.query({
+          name: permissionName,
+        });
+        if (permissionState.state !== 'granted') {
+          permissionState.addEventListener('change', handlePermissionChange);
+        } else {
+          enableSubscription(true);
```

Comment:
I think we should register the change event handler if the state is granted, as it can be revoked anytime

Reply:
Thank you Zita, I have fixed the code